pkg/frontend/transport/handler_test.go

```diff
@@ -164,6 +167,33 @@ func TestHandler_ServeHTTP(t *testing.T) {
 			activities, err := activitytracker.LoadUnfinishedEntries(activityFile)
 			require.NoError(t, err)
 			require.Empty(t, activities)
+
+			if tt.cfg.QueryStatsEnabled {
+				require.Len(t, logger.logMessages, 1)
+
+				msg := logger.logMessages[0]
+				require.Equal(t, "query stats", msg["msg"])
+				require.Equal(t, "query-frontend", msg["component"])
+				require.Equal(t, req.Method, msg["method"])
+				require.Equal(t, req.URL.Path, msg["path"])
+				require.Equal(t, req.UserAgent(), msg["user_agent"])
+				require.Contains(t, msg, "response_time")
+				require.Contains(t, msg, "query_wall_time_seconds")
+				require.EqualValues(t, 0, msg["fetched_series_count"])
+				require.EqualValues(t, 0, msg["fetched_chunk_bytes"])
+				require.EqualValues(t, 0, msg["fetched_chunks_count"])
+				require.EqualValues(t, 0, msg["fetched_index_bytes"])
+				require.EqualValues(t, 0, msg["sharded_queries"])
+				require.EqualValues(t, 0, msg["split_queries"])
+
+				for name, values := range tt.expectedParams {
+					logMessageKey := fmt.Sprintf("param_%v", name)
+					expectedValues := strings.Join(values, ",")
+					require.Equal(t, expectedValues, msg[logMessageKey])
+				}
+			} else {
+				require.Empty(t, logger.logMessages)
+			}
```

Comment:
Minor nit:
The `else` block looks like this test intends to make sure that the logged values are exactly the expected values and that there are no additional ones in addition to the expected values, but the `if` block only checks whether all the expected values are present and not if there are no additional values. For the check to be consistent maybe the `if` block should check the `len(msg)` as well?

Reply:
Good point, I've added this in 881406a.